### ISSUE TITLE
"problem.final_target_hit" returns incorrect value

### ISSUE THREAD

**Issue by asmaatamna** [numbbo/coco#1400]
When I type the following in an iPython shell:

```
In [1]: import cocoex

In [2]: suite = cocoex.Suite('bbob-constrained', '', '')

In [3]: problem = suite.get_problem_by_function_dimension_instance(7, 2, 1)

In [4]: problem.final_target_hit
Out[4]: 0

In [5]: problem([1e6, 2e6])
Out[5]: 4.6521438693022403e+18

In [6]: problem.final_target_hit
Out[6]: 1
```
I get one, meaning that the final target was hit, while the best f-value for this particular instance is ```2.207103403235e+06```from the .dat files.

**Comment by asmaatamna** [comment]
The problem above seems to occurs only with the ```bbob-constrained``` testbed. The same code with the ```bbob``` testbed gives the correct result.

```
In [7]: suite = cocoex.Suite('bbob', '', '')

In [8]: problem = suite.get_problem_by_function_dimension_instance(7, 2, 1)

In [9]: problem.final_target_hit
Out[9]: 0

In [10]: problem([1e6, 2e6])
Out[10]: 192096726128527.84

In [11]: problem.final_target_hit
Out[11]: 0
```

**Comment by brockho** [comment]
The problem seems to be the wrong bookkeeping (see #1393). I am still on the fix.

**Comment by nikohansen** [comment]
Closing as it will be fixed with addressing #1393
